### DISCUSSION TITLE
Generate Reactive REST server classes from OpenAPI yaml definitions

### DISCUSSION THREAD

**Discussion by gnieser** [quarkusio/quarkus#33717] [Q&A]
Hello,

For several years I've been using openapi-generator-maven-plugin jaxrs-spec with quarkus library to create server-side DTO and API classes from openapi.yaml files (in a contract-first approach).
I'm looking at the equivalent for reactive Quarkus with little success:
* I haven't found any openapi-generators for reactive Quarkus
* The quarkiverse openapi-generator is for client only
* The quarkiverse openapi-generator directs to apicurio-codegen for server side. Unfortunately, there is little documentation about it and I was not able to get it to work with my existing openapi.yaml files.

Has anyone some feedback about generating reactive REST server-side classes from openapi.yaml files?
Thanks

[a comment by an automated account is not shown]

**Comment by MikeEdgar** [comment]
Hi @gnieser , there is active development for this going in the https://github.com/Apicurio/apicurio-codegen project. It's still a work-in-progress, but depending on your timelines you may find it useful for this.

**Reply by gnieser** [reply]
Hi @MikeEdgar, thanks for taking time to answer.
I gave a try to apicurio-codegen, but somehow didn't found a way to make it generate anything. I do understand it is work-in-progress and I am tracking the project expecting good things. 
About timelines, the lack of such generator is putting an halt to my switch to reactive.
